tools/cldr-code/src/main/java/org/unicode/cldr/util/XPathParts.java

```diff
@@ -1273,4 +1262,14 @@ private XPathParts removeAttribute(String attribute) {
         }
         return this;
     }
+
+    public List<String> getPathAttributes() {
+        List<String> attributes = new ArrayList<>();
+        for (int i = 0; i < size(); ++i) {
+            for (String key : getAttributeKeys(i)) {
+                attributes.add(getAttributeValue(i, key));
+            }
```

Comment:
I think could be slightly simpler/faster as
```suggestion
            attributes.addAll(getAttributes().values());
```

Reply:
that is, `attributes.addAll(getAttributes(i).values());`

Reply:
Done in the 9th commit

Reply:
Yup. thanks